### PR TITLE
Add option to rotate SpriteWidget

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

Add option to rotate `SpriteWidget`

image

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

**Review by spydon — APPROVED** [review]
Lgtm!

**Review by erickzanardo — APPROVED** [review]

File: packages/flame/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
  - Take in an optional `Camera` as a parameter to `FlameGame`
  - It's no longer necessary to call `ComponentSet.register()` before `.query()`
  - Make super.onLoad mandatory to avoid memory leaks
+ - Add option to rotate SpriteWidget
```
Comment (wolfenrain) [inline packages/flame/CHANGELOG.md:11]:
```suggestion
 - Add option to rotate `SpriteWidget`
```

File: packages/flame/lib/src/widgets/sprite_widget.dart [hunk 2]
```diff
@@ -78,15 +84,19 @@ class _SpriteWidget extends StatelessWidget {
...
+
   const _SpriteWidget({
     required this.sprite,
     this.anchor = Anchor.topLeft,
+    this.angle = 0,
   });
 
   @override
   Widget build(_) {
     return Container(
```
Comment (wolfenrain) [inline packages/flame/lib/src/widgets/sprite_widget.dart:98]: Never dawned to me before, but why are we wrapping this in a container?

**Review by wolfenrain — APPROVED** [review]
LGTM, minor nits

File: packages/flame/lib/src/widgets/sprite_widget.dart (hunk 2 above)
Comment (luanpotter) [reply packages/flame/lib/src/widgets/sprite_widget.dart:98]: no idea...

Comment (luanpotter) [reply packages/flame/lib/src/widgets/sprite_widget.dart:98]: @erickzanardo will know

**Review by wolfenrain — APPROVED** [review]

**Pull request merged**